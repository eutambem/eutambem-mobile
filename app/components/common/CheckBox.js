import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet } from 'react-native';

import ExternalCheckBox from 'react-native-check-box';
import { DARK_PURPLE } from '../../styles';

const styles = StyleSheet.create({
  checkbox: {
    paddingTop: 10,
  },
});

const CheckBox = props => (
  <ExternalCheckBox
    {...props}
    style={styles.checkbox}
    checkBoxColor={DARK_PURPLE}
    rightText={props.label}
  />
);

CheckBox.propTypes = {
  label: PropTypes.string.isRequired,
};

export default CheckBox;
