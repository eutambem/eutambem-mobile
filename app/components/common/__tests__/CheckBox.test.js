import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import CheckBox from '../CheckBox';

const props = {
  label: 'Test CheckBox',
  onClick: sinon.spy(),
};

describe('CheckBox', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<CheckBox {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
