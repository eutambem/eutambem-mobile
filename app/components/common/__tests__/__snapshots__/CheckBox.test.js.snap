// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox renders correctly 1`] = `
<CheckBox
  checkBoxColor="#961687"
  isChecked={false}
  isIndeterminate={false}
  label="Test CheckBox"
  leftTextStyle={Object {}}
  onClick={[Function]}
  rightText="Test CheckBox"
  rightTextStyle={Object {}}
  style={
    Object {
      "paddingTop": 10,
    }
  }
/>
`;
